### PR TITLE
Enhancement: Performance improvements

### PR DESCRIPTION
**Pull request by smoench**
Bildschirmfoto 2020-10-14 um 17 39 51

### CODE REVIEW

File: src/Value/Line.php [hunk 1]
```diff
@@ -40,6 +46,24 @@ public function raw(): string
         return $this->raw;
     }
 
+    public function rawU(): UnicodeString
```
Comment (OskarStark) [inline src/Value/Line.php:49]: What about adding a parameter to `raw()` method `raw($asUnicodeString = false)` ?

Same for `clean()` 🤔 

Or rename `cleanU()` -> `cleanAsUnicodeString()` ? Imo the `U`-suffix is not very clear 🤔

File: src/Value/Lines.php [hunk 2]
```diff
@@ -58,7 +58,9 @@ public function next(): void
 
     public function key(): int
     {
-        $this->validate();
+        if (!isset($this->array[$this->currentLine])) {
```
Comment (OskarStark) [inline src/Value/Lines.php:61]: I would prefer `array_key_exists()` instead of `isset()` (in general)

File: src/Value/Line.php (hunk 1 above)
Comment (smoench) [reply src/Value/Line.php:49]: Then I would prefer using `*asUnicodeString()`

File: src/Value/Lines.php (hunk 2 above)
Comment (smoench) [reply src/Value/Lines.php:61]: `isset` is faster and the array holds only objects

File: src/Value/Line.php (hunk 1 above)
Comment (smoench) [reply src/Value/Line.php]: `clean` and `raw` now returns a `UnicodeString`-object

File: src/Helper/PhpHelper.php [hunk 3]
```diff
@@ -18,31 +18,11 @@
...
-
-        return false;
-    }
-
-    public static function isStartingWithOneBackslash(string $string): bool
-    {
-        return (bool) preg_match('/^[\\\\]{1}\w+/', $string);
-    }
-
-    public static function isStartingWithTwoBackslashes(string $string): bool
```
Comment (smoench) [inline src/Helper/PhpHelper.php:38]: unused

File: src/Helper/PhpHelper.php [hunk 4]
```diff
@@ -18,31 +18,11 @@
...
     public static function isComment(Line $line): bool
     {
-        if (preg_match('/^(#|\/\/)(.*)/', $line->clean())) {
-            return true;
-        }
-
-        return false;
-    }
-
-    public static function isStartingWithOneBackslash(string $string): bool
```
Comment (smoench) [inline src/Helper/PhpHelper.php:33]: unused

**Comment by OskarStark** [comment]
Thank you @smoench 🎉

**Pull request merged**